src/main/java/de/adesso/blogpostchecker/CheckExecutor.java

```diff
@@ -0,0 +1,97 @@
+
+    public void executeChecks(PostHeader postHeader, String authors) {
+        checkCategoryOnlyOneEntry(postHeader);
+        checkTagsNotEmpty(postHeader);
+        checkAuthorNotEmpty(postHeader);
+        checkAuthorInYml(postHeader, authors);
+        checkTitleNotEmpty(postHeader);
+        checkLayoutCorrect(postHeader);
+        checkDateMatchesFormat(postHeader);
+    }
+
+    private void checkCategoryOnlyOneEntry(PostHeader header) {
+        if (header.getCategories() != null && !header.getCategories().equals("")) {
+            if (!header.getCategories().contains(",")) {
+                LOGGER.info("Categories checked");
+            } else {
+                LOGGER.error("There are two or more categories.");
+                LOGGER.error("Exiting BlogpostChecker.");
+                System.exit(31);
+            }
+        } else {
+            LOGGER.error("Categories does not match.");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(30);
+        }
+    }
+
+    private void checkTagsNotEmpty(PostHeader header) {
+        if (header.getTags() != null && !header.getTags().equals("")) {
+            LOGGER.info("Tags checked");
+        } else {
+            LOGGER.error("Tags are empty");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(32);
+        }
+    }
+
+    private void checkAuthorNotEmpty(PostHeader header) {
+        if (header.getAuthor() != null && !header.getAuthor().equals("")) {
+            LOGGER.info("Author checked");
+        } else {
+            LOGGER.error("Author is empty.");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(33);
+        }
+    }
+
+    private void checkAuthorInYml(PostHeader header, String authors) {
+        if (authors.contains(header.getAuthor())) {
+            LOGGER.info("Author is in authors.yml.");
+        } else {
+            LOGGER.error("Author not in authors.yml checked.");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(34);
+        }
+    }
+
+    private void checkTitleNotEmpty(PostHeader header) {
+        if (header.getTitle() != null && !header.getTitle().equals("")) {
+            LOGGER.info("Title checked.");
+        } else {
+            LOGGER.error("Title is empty.");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(35);
+        }
+    }
+
+    private void checkLayoutCorrect(PostHeader header) {
+        if (header.getLayout() != null && header.getLayout().equals("post, post-xml")) {
+            LOGGER.info("Layout checked.");
+        } else {
+            LOGGER.error("Layout does not match 'post, post-xml'.");
+            LOGGER.error("Exiting BlogpostChecker.");
+            System.exit(36);
+        }
+    }
+
+    private void checkDateMatchesFormat(PostHeader header) {
+        if (header.getDate() != null && header.getDate().matches("\\d{4}-\\d{2}-\\d{2}\\s\\d{2}:\\d{2}")) {
```

Comment:
An der Stelle wäre die Extraktion in eine eigene Methode definitiv sinnvoll. Die sollte dann im Namen tragen welches Format das Datum trägt. Somit ergibt sich auch ein definitiver Grund die anderen zu extrahieren :)